sklearn/feature_selection/univariate_selection.py

```diff
@@ -266,17 +266,27 @@ def f_regression(X, y, center=True):
     f_classif: ANOVA F-value between label/feature for classification tasks.
     chi2: Chi-squared stats of non-negative features for classification tasks.
     """
-    if issparse(X) and center:
-        raise ValueError("center=True only allowed for dense data")
     X, y = check_X_y(X, y, ['csr', 'csc', 'coo'], dtype=np.float64)
+    n_samples = X.shape[0]
+
+    # compute centered values
+    # note that E[(x - mean(x))*(y - mean(y))] = E[x*(y - mean(y))], so we
```

Comment:
This comment is applicable only when y is mean centered correct? In which case it would be `E[x*y]`? (Sorry if I'm misunderstanding)